Add new function to isolate desired ROI

dschodt opened the pull request:
This branch adds a new function isolateROI() which lets you isolate a sub-ROI of an `smld` structure.

kalidke reviewed (changes requested):
Add example under the 'Utilities' section of main README

dschodt commented:
I've added an example to README.md and updated documentation.  I don't know how to make a meaningful generic method for the SMLD type so for now I've only changed the documentation for the SMLD2D method.
